src/ci/github-actions/calculate-job-matrix.py

```diff
@@ -44,22 +45,51 @@ def add_base_env(jobs: List[Job], environment: Dict[str, str]) -> List[Job]:
     return jobs
 
 
-class WorkflowRunType(enum.Enum):
-    PR = enum.auto()
-    Try = enum.auto()
-    Auto = enum.auto()
+@dataclasses.dataclass
+class PRRunType:
+    pass
+
+
+@dataclasses.dataclass
+class TryRunType:
+    custom_jobs: List[str]
+
+
+@dataclasses.dataclass
+class AutoRunType:
+    pass
+
+
+WorkflowRunType = typing.Union[PRRunType, TryRunType, AutoRunType]
 
 
 @dataclasses.dataclass
 class GitHubCtx:
     event_name: str
     ref: str
     repository: str
+    commit_message: Optional[str]
+
+
+def get_custom_jobs(ctx: GitHubCtx) -> List[str]:
+    """
+    Tries to parse names of specific CI jobs that should be executed in the form of
+    ci-job: <job-name>
+    from the commit message of the passed GitHub context.
+    """
+    if ctx.commit_message is None:
+        return []
+
+    regex = re.compile(r"ci-job: (.*)")
```

Comment:
(minor) For robustness:

```suggestion
    regex = re.compile(r"\bci-job: (.*)")
```

or better yet splitting the message by lines and then only accepting it at the very start of a line with `^` or `\A` to avoid triggering it in unexpected contexts, e.g., when quoting it with backticks.

Reply:
Good point. I think that `^` should be enough? The regex should be single-line by default.

Reply:
Right, but single-line means the opposite of what I was thinking. I do indeed need `re.MULTILINE`.